### ISSUE TITLE
UseFetcher causes links to be loaded again

### ISSUE THREAD

**Issue by s446683009** [remix-run/remix#5448]
### What version of Remix are you using?

1.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

add link.  
```javascript
/*root.tsx*/
import {cssBundleHref} from '@remix-run/css-bundle';
import iconfontCss from './styles/iconfont/iconfont.css'
import globalCss from './styles/global.css'

export const links = () => {
  return [
    {
      rel: "stylesheet",
      href: iconfontCss,
    },
    {
      rel: "stylesheet",
      href: globalCss,
    },
    ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
  
  ];
};

```
create  user login api route.  
```javascript
/*routes/apis_.user.login.tsx*/
import { ActionArgs } from "@remix-run/server-runtime";
export async function action({}:ActionArgs){
    return await new Promise((resolve,reject)=>{
        setTimeout(() => {
            resolve("login success");
        }, 3000);
    })
}


```
use `useFetcher` .
```JavaScript
/*routes/login.tsx*/
function Login() {
var fet=useFetcher();
useEffect(()=>{
     fet.submit({},{
      method:'post',
      action:'/apis/user/login',
   
  })},[]);


/*....*/
 }
```

### Expected Behavior

just load one time css link.  like this:
![image](https://user-images.githubusercontent.com/37138998/218643317-644f17e9-6537-48af-97e5-97776d5acef2.png)

### Actual Behavior

after fetch.submit api route. the css link in root.tsx load again.
![image](https://user-images.githubusercontent.com/37138998/218642932-65f86576-3c91-4db2-90ca-24e4041d043a.png)

by the way, I used css module.

**Comment by vaynevayne** [comment]
welcome nextjs

**Comment by brophdawg11** [comment]
This is fixed by #6679 and should be available in the next release.
